zproject/prod_settings_template.py

```diff
@@ -266,6 +266,27 @@
 #    ]
 # }
 
+## These settings can be used to synchronize ldap user groups to Zulip user groups
+## The name of the groups have to be exactly the same!
+## `cn` value of the ldap group is used
+
+## make sure to set `AUTH_LDAP_BIND_DN`, otherwise Zulip cannot access the groups
+# AUTH_LDAP_GROUP_SEARCH = LDAPSearch(
+#     "ou=groups,dc=www,dc=example,dc=com", ldap.SCOPE_SUBTREE, "(objectClass=groupOfUniqueNames)"
+# )
+
+# LDAP_GROUP_SYNC_LIST = {
+# 2 : [ # <-- 2 is in most cases the correct realm id, use `manage.py list_realms` to check!
+#   "group1",
+#   "group2",
+# ]
+# }
+
+## The type of ldap group. If the zulip group name IS NOT the value of `cn`,
+## then derive this class and overwrite the `def group_name_from_info(self, group_info)` function
```

Comment:
if "cn", also "common name" is an ldap attribute that is used by django as default group name. In some applications, e.g. the "description" attribute wants to be used. In that case one would do:

```
class CustomGroupType(GroupOfUniqueNamesType):
    def __init__(self):
        self.__init__(name_attr="description")

AUTH_LDAP_GROUP_TYPE = CustomGroupType()
```

alternatively

```
class CustomGroupType(GroupOfUniqueNamesType):
    def group_name_from_info(self, group_info):
        return group_info[1]["description"][0]

AUTH_LDAP_GROUP_TYPE = CustomGroupType()
```

Oh I also realized that this would be enough probably:

```
AUTH_LDAP_GROUP_TYPE = CustomGroupType(name_attr="description")
```

see this file: https://github.com/django-auth-ldap/django-auth-ldap/blob/master/django_auth_ldap/config.py#L437

the first option would give more flexibility (e.g. replacing _ with " " etc)

Reply:
I added a more comprehensive description: https://github.com/zulip/zulip/pull/23039/commits/aa00fd0d56c7579aa2170526715f5a8045251fa8